.github/workflows/pipeline.yml

```diff
@@ -16,11 +17,38 @@ jobs:
       - name: Setup Elixir
         uses: ./.github/actions/elixir-setup
         with:
-          elixir-version: "1.15.6" # Define the elixir version [required]
-          otp-version: "26.1" # Define the OTP version [required]
+          elixir-version: "1.15.6"
+          otp-version: "26.1"
+
+      - name: Install Dependencies
+        run: mix deps.get
+
+      - name: Run tests
+        run: mix test
+
+  publish:
+    name: Publish to hex.pm
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'created'
```

Comment:
If this has its own trigger, it would be more straightforward just to make a new workflow file triggered the normal way.